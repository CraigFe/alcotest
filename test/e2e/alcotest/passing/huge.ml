module Alcotest = Alcotest.V1

let pid = Unix.getpid ()

let read_all_lines ic =
  let acc = ref [] in
  try
    while true do
      acc := input_line ic :: !acc
    done;
    assert false
  with End_of_file -> String.concat "\n" !acc

let test_case n () =
  let open_fds =
    let ic =
      Unix.open_process_in ("lsof -p " ^ string_of_int pid ^ " 2>/dev/null")
    in
    let t = read_all_lines ic in
    close_in ic;
    t
  in
  Printf.printf "Test #%d. File descriptors open: %s\n%!" n open_fds

let () =
  Alcotest.run __FILE__
    [
      ( "alpha",
        List.init 10 (fun i ->
            Alcotest.test_case (string_of_int i) `Quick (test_case i)) );
    ]
