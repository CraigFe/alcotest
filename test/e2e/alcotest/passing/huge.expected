Testing `test/e2e/alcotest/passing/huge.ml'.
This run has ID `<uuid>'.

  [PASS]    alpha › 0
  [PASS]    alpha › 1
  [PASS]    alpha › 2
  [PASS]    alpha › 3
  [PASS]    alpha › 4
  [PASS]    alpha › 5
  [PASS]    alpha › 6
  [PASS]    alpha › 7
  [PASS]    alpha › 8
  [PASS]    alpha › 9

Full test results in `<build-context>/_build/_tests/<test-dir>'.
Test Successful in <test-duration>s. 10 tests run.

